pkg/generate/app/errors.go

```diff
@@ -53,3 +55,13 @@ The argument %[1]q could apply to the following Docker images or OpenShift image
 // ErrNameRequired is the error returned by new-app when a name cannot be
 // suggested and the user needs to provide one explicitly.
 var ErrNameRequired = fmt.Errorf("you must specify a name for your app with --name")
+
+// CircularOutputReferenceError is the error returned by new-app when the input
+// and output image stream tags are identical.
+type CircularOutputReferenceError struct {
+	ImageReference imageapi.DockerImageReference
+}
+
+func (e CircularOutputReferenceError) Error() string {
+	return fmt.Sprintf("the input and output Docker image references are identical (%q), please specify a different output reference with --to", e.ImageReference)
```

Comment:
--to is not generic to app builders - instead, newapp code needs to transform an error of type CircularOutputReferenceError into a message with `--to`.